benchmarks/harness/controller.py

```diff
@@ -278,6 +283,13 @@ def verifyStats(self):
     if args.disable_lsan:
         asan_options.append("detect_leaks=0")
 
+    # GPU tests require extra ASAN flags due to incompatibility with CUDA
+    # See: https://github.com/google/sanitizers/issues/629
+    if args.gpu is not None:
+        asan_options.extend(
+            ["protect_shadow_gap=0", "replace_intrin=0", "detect_leaks=0"]
```

Comment:
You're adding these here, but also in the JSON file. We need to pick a place and stick with it.

Reply:
My view is that JSON allows you to add these (and other flags) more on per need basis to individual tests.
However, these sanitizers have to be disabled otherwise you can't run CUDA kernels at all. Thus, they are here in controller as default options for GPU runs.